internal/resources/tanzukubernetescluster/schema.go

```diff
@@ -0,0 +1,402 @@
+	Description: "Control plane specific configuration.",
+	MaxItems:    1,
+	Required:    true,
+	Elem: &schema.Resource{
+		Schema: map[string]*schema.Schema{
+			ReplicasKey:    ReplicasSchema,
+			OSImageKey:     OSImageSchema,
+			common.MetaKey: common.Meta,
+		},
+	},
+}
+
+var NodePoolSchema = &schema.Schema{
+	Type:        schema.TypeList,
+	Description: "(Repeatable Block) Node pool definition for the cluster.",
+	Required:    true,
+	Elem: &schema.Resource{
+		Schema: map[string]*schema.Schema{
+			NameKey: {
+				Type:        schema.TypeString,
+				Description: "Name of the node pool.",
+				Required:    true,
+			},
+			DescriptionKey: {
+				Type:        schema.TypeString,
+				Description: "Description for the node pool.",
+				Optional:    true,
+			},
+			SpecKey: NodePoolSpecSchema,
+		},
+	},
+}
+
+var NodePoolSpecSchema = &schema.Schema{
+	Type:        schema.TypeList,
+	Description: "Spec for the node pool.",
+	MaxItems:    1,
+	Required:    true,
+	Elem: &schema.Resource{
+		Schema: map[string]*schema.Schema{
+			WorkerClassKey: {
+				Type:        schema.TypeString,
+				Description: "The name of the machine deployment class used to create the node pool.",
+				Required:    true,
+			},
+			FailureDomainKey: {
+				Type:        schema.TypeString,
+				Description: "The failure domain the machines will be created in.",
+				Optional:    true,
+			},
+			OverridesKey: {
+				Type:                  schema.TypeString,
+				Description:           "Overrides can be used to override cluster level variables.",
+				Optional:              true,
+				ValidateDiagFunc:      validateJSONString,
+				DiffSuppressOnRefresh: true,
+				DiffSuppressFunc:      checkVariablesValues,
+			},
+			ReplicasKey:    ReplicasSchema,
+			OSImageKey:     OSImageSchema,
+			common.MetaKey: common.Meta,
+		},
+	},
+}
+
+var ReplicasSchema = &schema.Schema{
+	Type:        schema.TypeInt,
+	Description: "Number of replicas",
+	Required:    true,
+}
+
+var OSImageSchema = &schema.Schema{
+	Type:        schema.TypeList,
+	Description: "OS image block",
+	MaxItems:    1,
+	Optional:    true,
+	Elem: &schema.Resource{
+		Schema: map[string]*schema.Schema{
+			NameKey: {
+				Type:        schema.TypeString,
+				Description: "The name of the OS image.",
+				Required:    true,
+			},
+			VersionKey: {
+				Type:        schema.TypeString,
+				Description: "The version of the OS image.",
+				Required:    true,
+			},
+			OSArchKey: {
+				Type:        schema.TypeString,
+				Description: "The architecture of the OS image.",
+				Required:    true,
+			},
+		},
+	},
+}
+
+func validateJSONString(value interface{}, path cty.Path) diag.Diagnostics {
+	valueJSON := make(map[string]interface{})
+	err := json.Unmarshal([]byte(value.(string)), &valueJSON)
+
+	if err != nil {
+		return diag.FromErr(errors.Wrapf(err, "Value is not a valid JSON string."))
+	}
+
+	return nil
+}
+
+func checkVariablesValues(k, oldValue, newValue string, d *schema.ResourceData) bool {
```

Comment:
Can a single variable change be reflected by `terrform plan`? I think the answer is yes, right? For example?
`+ variable_a = "xxx"`